src/cli/crd/generate.ts

```diff
@@ -0,0 +1,316 @@
+  kind: string | undefined;
+  fqdn: string;
+  scope: "Cluster" | "Namespaced";
+  plural: string;
+  shortNames?: string[];
+} {
+  const kind = extractSingleLineComment(content, "Kind");
+  const group = extractSingleLineComment(content, "Group") ?? "example";
+  const domain = extractSingleLineComment(content, "Domain") ?? "pepr.dev";
+  const details = extractDetails(sourceFile);
+
+  const fqdn = `${group}.${domain}`;
+
+  const { plural, scope } = details;
+
+  const shortNames = details.shortName ? [details.shortName] : undefined;
+  return { kind, plural, scope, shortNames, fqdn };
+}
+
+export async function generateCRDs(options: { output: string }): Promise<void> {
+  const outputDir = path.resolve(options.output);
+  await createDirectoryIfNotExists(outputDir);
+
+  const project = new Project();
+  const apiRoot = path.resolve("api");
+  const versions = getAPIVersions(apiRoot);
+
+  for (const version of versions) {
+    const sourceFiles = loadVersionFiles(project, path.join(apiRoot, version));
+    for (const sourceFile of sourceFiles) {
+      processSourceFile(sourceFile, version, outputDir);
+    }
+  }
+}
+
+export function getAPIVersions(apiRoot: string): string[] {
+  return fs.readdirSync(apiRoot).filter(v => fs.statSync(path.join(apiRoot, v)).isDirectory());
+}
+
+export function loadVersionFiles(project: Project, versionDir: string): SourceFile[] {
+  const files = fs.readdirSync(versionDir).filter(f => f.endsWith(".ts"));
+  const filePaths = files.map(f => path.join(versionDir, f));
+  return project.addSourceFilesAtPaths(filePaths);
+}
+
+export function processSourceFile(
+  sourceFile: SourceFile,
+  version: string,
+  outputDir: string,
+): void {
+  const content = sourceFile.getFullText();
+  const { kind, fqdn, scope, plural, shortNames } = extractCRDDetails(content, sourceFile);
+
+  if (!kind) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing '// Kind: <KindName>' comment`);
+    return;
+  }
+
+  const spec = sourceFile.getInterface(`${kind}Spec`);
+  if (!spec) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing interface ${kind}Spec`);
+    return;
+  }
+
+  const condition = sourceFile.getTypeAlias(`${kind}StatusCondition`);
+  const specSchema = getSchemaFromType(spec);
+  const conditionSchema = condition ? getSchemaFromType(condition) : emptySchema();
+
+  const crd = buildCRD({
+    kind,
+    fqdn,
+    version,
+    plural,
+    scope,
+    shortNames,
+    specSchema,
+    conditionSchema,
+  });
+
+  const outPath = path.join(outputDir, `${kind.toLowerCase()}.yaml`);
+  fs.writeFileSync(outPath, toYAML(crd), "utf8");
+  console.log(`✔ Created ${outPath}`);
+}
+
+// Extracts a comment from the content of a file based on a label.
+export function extractSingleLineComment(content: string, label: string): string | undefined {
+  // https://regex101.com/r/oLFaHP/1
+  const match = content.match(new RegExp(`//\\s+${label}:\\s+(.*)`));
+  return match?.[1].trim();
```

Comment:
Is the index `1` because `0` is something outside the capture group? At first glance this return statement felt like "return the second match" which made me curious.

Reply:
it is for extract certain details, like `kind`, `group`, `plurals`

The comments look like this in the generated file:

```ts
// Kind: Memcache
// Kind: VirtualService
// Kind: Gateway
// Kind: PeprStore
```